### PR TITLE
Development

### PR DESCRIPTION
**Pull request by em-em-D**
attendees route process get action , instead of Post action on the controller action,
am working alone on the app.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      session[:user_id]= user.id
+      redirect_to '/dashboard'
+    else
+      flash[:register_errors] = user.errors.full_messages
+      redirect_to '/'
+    end
+  end
+
+  def show
+    event = Event.all
```
Comment (efrapp) [inline app/controllers/users_controller.rb]: If you want to use the `event` variable in the show view you need to use an instance variable, see #5.

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Event < ApplicationRecord
```
Comment (efrapp) [inline app/models/event.rb]: Please add `scope` to get previous and upcoming events

File: app/controllers/attendees_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class AttendeesController < ApplicationController
+  before_action :logged_in_user
+
+  def attend
```
Comment (efrapp) [inline app/controllers/attendees_controller.rb]: It is better to use conventions to name the controller's actions, for example, if you are going to attend an event, it sounds like you want to `create` something.

File: app/controllers/attendees_controller.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    @event = Event.find(params[:event_id])
+    current_user.attend @event
+    flash[:success] = 'Successfully attended event'
+    redirect_to events_path
+  end
+
+  def leave
```
Comment (efrapp) [inline app/controllers/attendees_controller.rb]: Same here.

File: app/controllers/events_controller.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        @events = Event.all
+        @future_events = @events.future_events
+        @past_events = @events.past_events
+    end
+    def new
+        @event = Event.new
+    end
+    
+    def create
+        @event = current_user.events.build(event_params)
```
Comment (SunnyAsar) [inline app/controllers/events_controller.rb]: Screenshot 2019-08-27 at 5 35 11 PM

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      session[:user_id]= @user.id
```
Comment (SunnyAsar) [inline app/controllers/users_controller.rb]: Screenshot 2019-08-27 at 5 34 30 PM

File: test/controllers/attendees_controller_test.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
+require 'test_helper'
+
+class AttendeesControllerTest < ActionDispatch::IntegrationTest
+  test "should get index" do
+    get attendees_index_url
+    assert_response :success
+  end
+
```
Comment (SunnyAsar) [inline test/controllers/attendees_controller_test.rb]: Screenshot 2019-08-27 at 5 32 41 PM

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @attended_events = @user.attended_events
+    @events = @user.events
+  end
```
Comment (efrapp) [inline app/controllers/users_controller.rb]: I don't see you are showing the previous and upcoming events attended for an specific user. Also, check this error when I try to check one user: #8

File: app/views/public/_eventstatus.html.erb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<% @user = current_user %>
+<% @event = event %>
```
Comment (efrapp) [inline app/views/public/_eventstatus.html.erb]: You don't need those assignments, `current_user` is a global helper and `event` is a variable that should come from the view that calls this partial.

File: config/routes.rb [hunk 10]
```diff
@@ -1,3 +1,31 @@
...
+  # get 'events/show'
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  
+  get '/signup',      to: 'users#new'
+  post '/signup',     to: 'users#create'
+  get '/login',       to: 'sessions#new'
+  post '/login',      to: 'sessions#create'
+  delete '/logout',   to: 'sessions#destroy'
+  root 'sessions#new'
+  get '/eventlist', to: 'events#index'
```
[the remark this answers is not in the document]
Comment (efrapp) [reply config/routes.rb]: You need to use `localhost:3000/events` to display all the events, don't use `eventlist` please. Also, when I try to click the `events` link in the navbar that link doesn't do anything.

File: app/views/public/_eventstatus.html.erb [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+<% if @user &.event? @event %>
+    <%= link_to 'Delete', event_path(@event), method: :delete, data: {confirm: "Are you sure you want to delete this event?"}, class: "action-btn" %>
+    <% end %>
+<% if @user &.attended_event? @event %>
+    <%= link_to 'Leave',  attendaces_path(event_id: @event.id), method: :delete, data: {confirm: "Leave this event?" }, class: "action-btn" %>
+<% end %>
+<% if @event.upcoming? %>
+    <%= link_to 'Attend', attendaces_path(event_id: @event.id), method: :post, data: {confirm: "Attend this event?" }, class: "action-btn" %>
```
Comment (efrapp) [inline app/views/public/_eventstatus.html.erb]: I get this error when I try to attend an event: #3.

Comment (em-em-D) [reply app/views/public/_eventstatus.html.erb]: hi, @efrapp I think that particular problem in attending an event from users/show. is from your end. probably not running "bundle update before starting the application" it works fine on my machine, I would update the code

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+      flash[:success] = 'You are signed up'
+      redirect_to '/login'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @attended_events = current_user.attended_events
```
[the remark this answers is not in the document]
Comment (em-em-D) [reply app/controllers/users_controller.rb]: I would do that

File: app/views/users/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+<h2><%= @user.username %> Events</h2>
+<ul>
+  <% @user.events.each do |event| %>
```
[the remark this answers is not in the document]
Comment (em-em-D) [reply app/views/users/show.html.erb]: ok

**Comment by em-em-D** [comment]
all requested changes have been made

**Pull request merged**